### PR TITLE
feat: show additional info for autocomplete results

### LINKED ISSUE #23381: [Feature]: Support autocomplete helper text to explain JS functions
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Add support to provide information about JS functions as seen on other IDEs. We unintentionally added this support during code optimization #21667 and code editor visual changes on #22652 but need to reintroduce it back with some design and UX alignment.

### Why should this be worked on?

Most of our users are new/beginners to Javascript and this feature would aid in providing help to developers. 
https://www.loom.com/share/a39a57ae79794860807d185704ef9178

This, along with #4789 & #10578 would help improve the coding experience.

### PR DESCRIPTION
**Pull request by arunvjn**
## Description
Adds another popover against a selected autocomplete result that shows relevant information beforehand to the user. Info includes a brief description, example if applicable and a link to docs.
>
#### PR fixes following issue(s)
Fixes #23381 
>
#### Media
Screenshot 2023-11-07 at 13 00 25

#### Type of change
- New feature (non-breaking change which adds functionality)
>
#### How Has This Been Tested?
- [x] Manual
- [x] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/ce/utils/autocomplete/EntityDefinitions.ts [hunk 1]
```diff
@@ -99,57 +137,89 @@ export const GLOBAL_DEFS = {
...
+      "Enables navigation between the internal pages of the App or to an external URL.",
     "!type":
       "fn(pageNameOrUrl: string, params: {}, target?: string) -> +Promise",
   },
   showAlert: {
-    "!doc": "Show a temporary notification style message to the user",
+    "!url":
+      "https://docs.appsmith.com/reference/appsmith-framework/widget-actions/show-alert",
+    "!doc":
+      "Displays a temporary toast-style alert message to the user for precisely 5 seconds. The duration of the alert message can't be modified.",
```
Comment (ohansFavour) [inline app/client/src/ce/utils/autocomplete/EntityDefinitions.ts:151]: This is well-detailed!. Nice!

File: app/client/src/ce/utils/autocomplete/EntityDefinitions.ts [hunk 2]
```diff
@@ -99,57 +137,89 @@ export const GLOBAL_DEFS = {
...
     "!doc": "Close a modal",
     "!type": "fn(modalName: string) -> +Promise",
   },
   storeValue: {
-    "!doc": "Store key value data locally",
-    "!type": "fn(key: string, value: any) -> +Promise",
+    "!url":
+      "https://docs.appsmith.com/reference/appsmith-framework/widget-actions/store-value",
+    "!doc":
+      "Stores the data in the browser's local storage as key-value pairs that represent storage objects and can be later accessed anywhere in the application.",
```
Comment (ohansFavour) [inline app/client/src/ce/utils/autocomplete/EntityDefinitions.ts:171]: "...can be later accessed anywhere in the application via `appsmith.store`".

Let's consider educating users on how to access the values too.

File: app/client/src/components/editorComponents/CodeEditor/styledComponents.ts [hunk 3]
```diff
@@ -37,6 +37,7 @@ export const CodeEditorColors = {
   NUMBER: "#555",
   COMMENT: "var(--ads-v2-color-gray-400)",
   FUNCTION_ARGS: "hsl(288, 44%, 44%)",
+  TOOLTIP_FN_ARGS: "#DB6E33",
```
Comment (ohansFavour) [inline app/client/src/components/editorComponents/CodeEditor/styledComponents.ts:40]: Can you confirm if we have an ADS variable for this color value?

File: app/client/src/constants/defs/browser.json [hunk 4]
```diff
@@ -245,12 +245,12 @@
   "setTimeout": {
     "!type": "fn(f: fn(), ms: number) -> number",
     "!url": "https://developer.mozilla.org/en/docs/DOM/window.setTimeout",
-    "!doc": "Calls a function or executes a code snippet after specified delay."
+    "!doc": "Calls a function or executes a code snippet after specified delay. Returns a number that can be used to clear the timeout with clearTimeout()."
```
Comment (ohansFavour) [inline app/client/src/constants/defs/browser.json:248]: Educating users on the use of "clearTimeout()" here is nice :thumbsup:

File: app/client/src/ce/utils/autocomplete/EntityDefinitions.ts [hunk 5]
```diff
@@ -22,43 +22,81 @@ export const entityDefinitions = {
...
+        "!type": "fn(params: ?) -> +Promise",
+        "!url":
+          "https://docs.appsmith.com/reference/appsmith-framework/query-object#queryrun",
+        "!doc": "Execute the query with the given parameters.",
+      },
+      clear: {
+        "!type": "fn() -> +Promise",
+        "!url":
+          "https://docs.appsmith.com/reference/appsmith-framework/query-object#queryclear",
+        "!doc": "Clear the query data.",
```
Comment (ohansFavour) [inline app/client/src/ce/utils/autocomplete/EntityDefinitions.ts:98]: `Clears the query data`

File: app/client/src/ce/utils/autocomplete/EntityDefinitions.ts [hunk 6]
```diff
@@ -22,43 +22,81 @@ export const entityDefinitions = {
...
+          "Object that contains details about the response, such as the status code, headers, and other relevant information related to the query's execution and the server's response.",
+        "!url":
+          "https://docs.appsmith.com/reference/appsmith-framework/query-object#responsemeta-object",
+        ...responseMetaDef,
+      },
+      run: {
+        "!type": "fn(params: ?) -> +Promise",
+        "!url":
+          "https://docs.appsmith.com/reference/appsmith-framework/query-object#queryrun",
+        "!doc": "Execute the query with the given parameters.",
```
Comment (ohansFavour) [inline app/client/src/ce/utils/autocomplete/EntityDefinitions.ts:92]: Let's consider using the same tense across the board. `Executes the query...`

**Review by ohansFavour — DISMISSED** [review]

File: app/client/src/components/editorComponents/CodeEditor/styledComponents.ts (hunk 3 above)
Comment (arunvjn) [reply app/client/src/components/editorComponents/CodeEditor/styledComponents.ts:40]: These colors are not a part of ADS. CodeEditor & its colors are not part of ADS today. Got a confirmation from Albin

**Review by ohansFavour — APPROVED** [review]

**Pull request merged**